File: hfsfolder-cpp/PreFile.cpp
#include "PreFile.h"

namespace hfsfolder_model
{

PreFile::PreFile()
{
}

PreFile::PreFile(std::string name, unsigned long long size, ptime modified, 
	std::string attributes, std::string formatSize, std::string formatModified) {
	this->name = name;
	this->size = size;
	this->modified = modified;
	this->attributes = attributes;
	this->formatedSize = formatSize;
	this->formatedModified = formatModified;
}
	
PreFile::~PreFile()
{
}

std::string PreFile::getName()
{
	return name;
}

void PreFile::setName(std::string name)
{
	this->name = name;
}

unsigned long long PreFile::getSize()
{
	return size;
}

void PreFile::setSize(unsigned long long size)
{
	this->size = size;
}

ptime PreFile::getModified()
{
	return modified;
}

void PreFile::setModified(ptime modified)
{
	this->modified = modified;
}

std::string PreFile::getAttributes()
{
	return attributes;
}

void PreFile::setAttributes(std::string attributes)
{
	this->attributes = attributes;
}

std::string PreFile::getFormatedSize()
{
	return formatedSize;
}

void PreFile::setFormatedSize(std::string formatedSize)
{
	this->formatedSize = formatedSize;
}

std::string PreFile::getFormatedModified()
{
	return formatedModified;
}

void PreFile::setFormatedModified(std::string formatedModified)
{
	this->formatedModified = formatedModified;
}

std::string PreFile::getOriginalPath() {
	return originalPath;
}

void PreFile::setOriginalPath(std::string originalPath) {
	this->originalPath = originalPath;
}

void PreFile::setDirectory(bool directory){
	this->directory = directory;
}

bool PreFile::isDirectory(){
	return directory;
}
	
void PreFile::limparDados() {
	this->name = "";
	this->size = 0;
	this->modified = second_clock::local_time();
	this->attributes = "";
	this->formatedSize = "";
	this->formatedModified = "";	
	this->originalPath = "";
}

std::string PreFile::toString() {
	std::stringstream ss;
	ss << "PreFile [name=" << name << ", size=" << size << ", modified=" << to_simple_string(modified) 
		<< ", attributes=" << attributes << ", formatSize=" << formatedSize 
		<< ", formatModified=" << formatedModified << "]";
	return ss.str();
}

std::string PreFile::toInsert() {
	std::stringstream ss;
	ss << "'" << getName() << "'," << size << ",'" << formatedModified << "','" << attributes << "'";
	return ss.str();
}

std::string PreFile::toCVS() {
	std::stringstream ss;
	ss << getName() << ";" << size << ";" << formatedModified << ";" << attributes;
	return ss.str();
}
	
} // namespace model

File: hfsfolder-php/model/PreFile.php
<?php


namespace model;


class PreFile
{
    private $name = "";
    private $size = 0;
    private $modified;
    private $attributes = "";
    private $formatedSize = "";
    private $formatedModified = "";
    private $originalPath = "";
    private $directory = false;

    public function __construct() {
        $this->limparDados();
    }

    /**
     * @return string
     */
    public function getName()
    {
        return $this->name;
    }

    /**
     * @param string $name
     */
    public function setName($name)
    {
        $this->name = $name;
    }

    /**
     * @return int
     */
    public function getSize()
    {
        return $this->size;
    }

    /**
     * @param int $size
     */
    public function setSize($size)
    {
        $this->size = $size;
    }

    /**
     * @return mixed
     */
    public function getModified()
    {
        return $this->modified;
    }

    /**
     * @param mixed $modified
     */
    public function setModified($modified)
    {
        $this->modified = $modified;
    }

    /**
     * @return string
     */
    public function getAttributes()
    {
        return $this->attributes;
    }

    /**
     * @param string $attributes
     */
    public function setAttributes($attributes)
    {
        $this->attributes = $attributes;
    }

    /**
     * @return string
     */
    public function getFormatedSize()
    {
        return $this->formatedSize;
    }

    /**
     * @param string $formatedSize
     */
    public function setFormatedSize($formatedSize)
    {
        $this->formatedSize = $formatedSize;
    }

    /**
     * @return string
     */
    public function getFormatedModified()
    {
        return $this->formatedModified;
    }

    /**
     * @param string $formatedModified
     */
    public function setFormatedModified($formatedModified)
    {
        $this->formatedModified = $formatedModified;
    }

    /**
     * @return string
     */
    public function getOriginalPath()
    {
        return $this->originalPath;
    }

    /**
     * @param string $originalPath
     */
    public function setOriginalPath($originalPath)
    {
        $this->originalPath = $originalPath;
    }

    /**
     * @return bool
     */
    public function isDirectory()
    {
        return $this->directory;
    }

    /**
     * @param bool $directory
     */
    public function setDirectory($directory)
    {
        $this->directory = $directory;
    }

    public function limparDados() {
        $this->name = "";
        $this->size = 0;
        $this->modified = date("d/m/Y H:i:s");
        $this->attributes = "";
        $this->formatedSize = "";
        $this->formatedModified = "";
        $this->originalPath = "";
        $this->directory = false;
    }

    public function __toString() {
        return "PreFile [name=" . $this->name . ", size=" . $this->size
            . ", modified=" . $this->modified . ", attributes=" . $this->attributes
            . ", formatSize=" . $this->formatedSize
            . ", formatModified=" . $this->formatedModified . "]";
    }

    public function toCVS() {
        return $this->getName() . ";" . $this->size . ";"
            . $this->formatedModified . ";" . $this->attributes;
    }

}